pkg/reconciler/broker/broker_test.go

```diff
@@ -376,13 +459,23 @@ func TestAllCases(t *testing.T) {
 			drStore = NewDataresidencyTestStore(t, cm.(*corev1.ConfigMap))
 		}
 
+		// if maxPSClientCreateTime is in testData, no pubsub client is passed to reconciler, the reconciler
+		// will create one in demand
+		testPSClient := psclient
+		if maxTime, ok := testData["maxPSClientCreateTime"]; ok {
+			// Overwrite the createPubsubClientFn to one that failed when calling more than once, this does
+			// not really work since we never test anything that reconcile twice or both reconcile and delete
+			createPubsubClientFn = GetFailedTestClientCreateFunc(srv.Addr, maxTime.(int))
+			testPSClient = nil
+		}
+
 		ctx = addressable.WithDuck(ctx)
 		ctx = resource.WithDuck(ctx)
 		r := &Reconciler{
 			Base:               reconciler.NewBase(ctx, controllerAgentName, cmw),
 			brokerCellLister:   listers.GetBrokerCellLister(),
 			projectID:          testProject,
-			pubsubClient:       psclient,
+			pubsubClient:       testPSClient,
```

Comment:
Should we also test the case where this is nil when the Reconciler is instantiated, as in https://github.com/google/knative-gcp/blob/42a25fe3dea6ea8f19d5d12607e3e1a5a59be98f/pkg/reconciler/broker/controller.go#L75?

Or maybe that's already tested elsewhere.

Reply:
This might also answer your previous question about why have the `maxPSClientCreateTime`. If you look at how testPSClient is set, when `maxPSClientCreateTime` exists it will set to nil instead of `psclient` so that the pubsub client will be created during reconciliation. That's why we can make sure the `pubsub.NewClient` function is called only once in the first test case and inject error in the second test case.

Reply:
Thanks! I missed that on the first read through.